lib/source-loader/src/build.js

```diff
@@ -1,24 +1,60 @@
+import { SourceMapConsumer, SourceNode } from 'source-map';
+
 import { readStory } from './dependencies-lookup/readAsObject';
 
-export function transform(inputSource) {
-  return readStory(this, inputSource).then((sourceObject) => {
+export async function transform(inputSource, inputSourceMap) {
+  const callback = this.async();
```

Comment:
🤯

Reply:
i've never seen this construct and am having trouble finding documentation about it. can you point me at the docs for this?

Reply:
Indeed rather confusing :) 

... but this is actually just [Webpack's Loader API](https://webpack.js.org/api/loaders/), which provides a pretty fancy "this" to work with. [`async`](https://webpack.js.org/api/loaders/#thisasync) marks the loader as "will call back asynchronously", and then returns a callback function. Previously just returning the transformed source didn't need any of that, but now the `transform` function needs to return both the source and the source map, so it needs to use this callback.

Reply:
Awesome, thanks for clarifying. I thought it was a native JS thing and my head nearly exploded.